packages/wallet/src/wallet.ts

```diff
@@ -213,63 +218,80 @@ export class DemexWallet {
     return await this.getDefaultSigningData(txRequest);
   }
 
-  private async checkReloadAccountState(signer: DemexSigner) {
+  private async initWalletAccountState(signer: DemexSigner) {
     const address = await getDefaultSignerAddress(signer);
-    const state = this.walletAccounts?.[address]
-    if (!state || state.sequenceInvalidated) return await this.reloadAccount(signer)
+    const walletAccount = this.walletAccounts?.[address];
+    if (!walletAccount) return await this.reloadAccountFromSigner(signer);
+  }
+
+  private async reloadAccountFromSigner(signer: DemexSigner) {
+    const { bech32Address, evmBech32Address } = await getDefaultSignerAddresses(signer, this.networkConfig.bech32Prefix);
+    await this.reloadAccount(bech32Address, evmBech32Address);
   }
 
   /**
   * Reloads primary account state as priority
   * Only tries to reload secondary account state if primary account state is not found 
-  * and signer implements interface to get secondary account address
+  * Returns the updated wallet account info
   */
-  public async reloadAccount(signer: DemexSigner) {
-    const info = await this.reloadAccountInfo(signer);
-    const address = await getDefaultSignerAddress(signer);
+  public async reloadAccount(bech32Address: string, evmBech32Address?: string) {
+    const info = await this.reloadAccountInfo(bech32Address, evmBech32Address);
     if (!info) return;
-    this.walletAccounts[address] = { ...info, sequenceInvalidated: false };
+    this.walletAccounts[bech32Address] = info;
+    await this.updateMergeAccountStatus(bech32Address);
+    return this.walletAccounts[bech32Address];
   }
 
-  private async reloadAccountInfo(signer: DemexSigner) {
-    const address = await getDefaultSignerAddress(signer);
-    const account: Account | undefined = await this.getAccount(address);
+
+  private async reloadAccountInfo(bech32Address: string, evmBech32Address?: string) {
```

Comment:
this "reload" in `reloadAccountInfo` implies state change, but in fact it does not.